ledger-service/http-json/src/itlib/scala/http/AbstractWebsocketServiceIntegrationTest.scala

```diff
@@ -857,6 +857,67 @@ abstract class AbstractWebsocketServiceIntegrationTest
     case _ => -\/(jsv)
   }
 
+  "no duplicates should be returned when retrieving contracts for multiple parties" in withHttpService {
+    (uri, encoder, _) =>
+      val aliceAndBob = List("Alice", "Bob")
+      val sharedAccountNumber = "4444"
+      val jwtForAliceAndBob = jwtForParties(actAs = aliceAndBob, readAs = Nil, ledgerId = testId)
+      import spray.json._
+
+      val (killSwitch, source) = singleClientQueryStream(
+        jwt = jwtForAliceAndBob,
+        serviceUri = uri,
+        query = """{"templateIds": ["Account:SharedAccount"]}""",
+      )
+        .viaMat(KillSwitches.single)(Keep.right)
+        .preMaterialize()
+
+      def test: Sink[JsValue, Future[ShouldHaveEnded]] = {
+        val dslSyntax = Consume.syntax[JsValue]
+        import dslSyntax._
+        Consume.interpret(
+          for {
+            ContractDelta(Vector(), _, Some(liveStartOffset)) <- readOne
+            (status, _) <- liftF(
+              postCreateCommand(
+                sharedAccountCreateCommand(aliceAndBob, sharedAccountNumber),
+                encoder,
+                uri,
+                headers = headersWithPartyAuth(aliceAndBob),
+              )
+            )
```

Comment:
Any `create` following the `liveStartOffset` message is _guaranteed_ not to go through SQL. To guarantee that a create _does_ go through SQL, you must sequence the `postCreateCommand` future to complete before the `runWith` step on line 915. So if you move your create down there instead of delaying it with the `Consume` language's `liftF`, you should be golden.

Reply:
Oh, I thought everything went through the database. This actually explains why breaking the query entirely (`WHERE 1 = 2`) didn't break the test. Thanks, I'll see if I can make it work tomorrow.

Reply:
Addressed in https://github.com/digital-asset/daml/pull/9443/commits/d3eb4fdf82b15164521ee49cc506d67afde14b46 and checked that the test does indeed catch a regression with `WHERE 1 = 1` as mentioned above. Thanks for double checking and clarifying how to make this work.